src/shared/hooks/useCreditGuard.ts

```
@@ -0,0 +1,97 @@
+ *
+ * @param requiredCredits 필요한 크레딧 수 (기본값: 1)
+ * @returns 크레딧 확인 함수와 상태 정보
+ */
+export const useCreditGuard = (
+  requiredCredits: number = 1
+): CreditGuardReturn => {
+  // 사용자 데이터 조회 (실시간으로 API 호출)
+  const { data: userData, isLoading, refetch } = useMyPageUser();
+
+  // 토스트 알림 훅
+  const { notify } = useToast();
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)